lib/sailor.js

```diff
@@ -61,15 +62,32 @@ class Sailor {
         return co(function* doStartup() {
             console.log('Starting up component');
             const result = yield this.invokeModuleFunction('startup');
-            //TODO: persist it
+
+            const handle = hooksData.startup(this.settings);
+            try {
+                const dataToStore = _.isEmpty(result) ? {} : result;
+                yield handle.create(dataToStore);
+            } catch (e) {
+                if (e.statusCode === 409) {
+                    console.warn('Hooks data creating conflict. Deleting existing data');
+                    yield handle.delete();
+                    yield handle.create(result);
```

Comment:
Fail will cause an exception, which will be caught by general error handler and sailor will exit with log message about error.